src/hotspot/share/code/nmethod.hpp

```diff
@@ -469,6 +469,62 @@ class nmethod : public CodeBlob {
   void oops_do_set_strong_done(nmethod* old_head);
 
 public:
+  enum class ChangeReason : u1 {
+    C1_codepatch,
+    C1_deoptimize,
+    C1_deoptimize_for_patching,
+    C1_predicate_failed_trap,
+    CI_replay,
+    JVMCI_invalidate_nmethod,
+    JVMCI_invalidate_nmethod_mirror,
+    JVMCI_materialize_virtual_object,
+    JVMCI_new_installation,
+    JVMCI_register_method,
+    JVMCI_replacing_with_new_code,
+    JVMCI_reprofile,
+    marked_for_deoptimization,
+    missing_exception_handler,
+    not_used,
+    OSR_invalidation_back_branch,
+    OSR_invalidation_for_compiling_with_C1,
+    OSR_invalidation_of_lower_level,
+    set_native_function,
+    uncommon_trap,
+    whitebox_deoptimization,
+    zombie,
+  };
+
+
+  static const char* change_reason_to_string(ChangeReason change_reason) {
+    switch (change_reason) {
+        case ChangeReason::C1_codepatch:                            return "C1 code patch";
```

Comment:
Indenting: should be two spaces everywhere. Also, I think this kind of indenting forces us to re-align the switch for the largest enum label. Let's just break them. Plus, any multi-line blocks should be braced. So, in total:

```
switch (change_reason) {
  case ChangeReason::C1_codepatch: 
    return "C1 code patch";
  ...
  default: {
    assert(false, "Unhandled reason");
    return "Unknown";
  }
}
```